box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFile.java

```diff
@@ -1206,4 +1221,379 @@ public void setRequestedPage(int page) {
             mRequestPage = page;
         }
     }
+    /**
+     * Request for creating an upload session for multiput upload
+     */
+    public static class CreateUploadSession extends BoxRequest<BoxUploadSession, CreateUploadSession> {
+        private static final long serialVersionUID = 8145675031279971502L;
+
+        String mDestinationFolderId;
+        String mFileName;
+        long mFileSize;
+        private FileInputStream mFileInputStream;
+
+        /**
+         * Creates an upload file session  with the default parameters
+         *
+         * @param fileName  name of the new file
+         * @param fileSize file size in bytes
+         * @param destinationFolderId   id of the parent folder for the new file
+         * @param requestUrl    URL of the upload file endpoint
+         * @param session   the authenticated session that will be used to make the request with
+         */
+        public CreateUploadSession(String fileName, long fileSize, String destinationFolderId, String requestUrl, BoxSession session) {
```

Comment:
If someone uses this constructor, what happens to the mFileInputStream, which is only instantiated on the other constructor ? Apparently the onSendComplete method will crash as it will be null.